pkg/odo/cli/preference/set_test.go

```diff
@@ -41,7 +41,8 @@ func TestSet(t *testing.T) {
 	}
 
 	prefClient.EXPECT().SetConfiguration("arg1", "Arg2")
-	err = opts.Run(cmdline)
+	cmdline.EXPECT().Context()
+	err = opts.Run(cmdline.Context())
```

Comment:
Or you could pass a `context.Background()` here

Reply:
I guess so but the actual use case would involve a command line context. Should we not try to be as close as possible to the real thing?

Reply:
updated

Reply:
`cmdline.EXPECT().Context()` will make the `Context()` method return the zero-value of the returned value, which is I think a nil pointer. Which is not what you want